sycl/include/sycl/ext/oneapi/matrix/matrix-jit-use.hpp

```diff
@@ -0,0 +1,517 @@
+                        PI_INVALID_DEVICE);
+#endif // __SYCL_DEVICE_ONLY__
+  }
+
+  inline __SYCL_ALWAYS_INLINE wi_slice<T, NumRows, NumCols, Use, Layout, Group>
+  get_wi_data() {
+    return wi_slice<T, NumRows, NumCols, Use, Layout, Group>(*this);
+  }
+};
+
+template <typename Group, typename T, size_t NumRows, size_t NumCols,
+          matrix_use Use = matrix_use::unnecessary,
+          matrix_layout Layout = matrix_layout::row_major,
+          access::address_space Space>
+inline __SYCL_ALWAYS_INLINE void
+joint_matrix_load(Group sg,
+                  joint_matrix<T, NumRows, NumCols, Use, Layout, Group> &res,
+                  multi_ptr<T, Space> src, size_t stride, matrix_layout MemL) {
+#ifdef __SYCL_DEVICE_ONLY__
+  T *Ptr = src.get();
+  switch (MemL) {
+  default:
+    assert(false && "Invalid Memory Layout!");
+  case matrix_layout::row_major:
+    res.spvm =
+        __spirv_JointMatrixLoadINTEL<T, NumRows, NumCols,
+                                     spv_matrix_use_traits<Use>::value,
+                                     spv_matrix_layout_traits<Layout>::value>(
+            Ptr, stride, __spv::MatrixLayout::RowMajor,
+            spv_scope_traits<Group>::value);
+    break;
+  case matrix_layout::col_major:
+    res.spvm =
+        __spirv_JointMatrixLoadINTEL<T, NumRows, NumCols,
+                                     spv_matrix_use_traits<Use>::value,
+                                     spv_matrix_layout_traits<Layout>::value>(
+            Ptr, stride, __spv::MatrixLayout::ColumnMajor,
+            spv_scope_traits<Group>::value);
+    break;
+  case matrix_layout::packed_a:
+    res.spvm =
+        __spirv_JointMatrixLoadINTEL<T, NumRows, NumCols,
+                                     spv_matrix_use_traits<Use>::value,
+                                     spv_matrix_layout_traits<Layout>::value>(
+            Ptr, stride, __spv::MatrixLayout::PackedA,
+            spv_scope_traits<Group>::value);
+    break;
+  case matrix_layout::packed_b:
+    res.spvm =
+        __spirv_JointMatrixLoadINTEL<T, NumRows, NumCols,
+                                     spv_matrix_use_traits<Use>::value,
+                                     spv_matrix_layout_traits<Layout>::value>(
+            Ptr, stride, __spv::MatrixLayout::PackedB,
+            spv_scope_traits<Group>::value);
+    break;
+  }
+#else
+  (void)sg;
+  (void)res;
+  (void)src;
+  (void)stride;
+  (void)MemL;
+  throw runtime_error("joint matrix is not supported on host device.",
+                      PI_INVALID_DEVICE);
+#endif // __SYCL_DEVICE_ONLY__
+}
+
+template <typename Group, typename T, size_t NumRows, size_t NumCols,
+          matrix_use Use = matrix_use::unnecessary,
+          matrix_layout MatL = matrix_layout::row_major,
+          access::address_space Space>
+inline __SYCL_ALWAYS_INLINE void
+joint_matrix_store(Group sg,
+                   joint_matrix<T, NumRows, NumCols, Use, MatL, Group> &src,
```

Comment:
This is a more interesting point:

Are you sure that the `matrix_use` is necessary as a parameter for `joint_matrix_store`? In the CUDA backend it is not necessary because you can only call `joint_matrix_store` for a `matrix_use::accumulator` matrix, since there is no use case to store a matrix back to a  matrix tile pointer storage other than the matrix that is the output of a MMA, which is always a `matrix_use::accumulator` matrix. I'd be surprised if this was different in the Intel backend!?